File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/operation/fetchoptions/OperationExecutionFetchOptions.java
/*
 * Copyright 2014 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package ch.ethz.sis.openbis.generic.asapi.v3.dto.operation.fetchoptions;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.fetchoptions.FetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.fetchoptions.FetchOptionsToStringBuilder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.operation.OperationExecution;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.operation.fetchoptions.OperationExecutionDetailsFetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.operation.fetchoptions.OperationExecutionNotificationFetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.operation.fetchoptions.OperationExecutionSummaryFetchOptions;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.person.fetchoptions.PersonFetchOptions;
import ch.systemsx.cisd.base.annotation.JsonObject;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.io.Serializable;

/*
 * Class automatically generated with DtoGenerator
 */
@JsonObject("as.dto.operation.fetchoptions.OperationExecutionFetchOptions")
public class OperationExecutionFetchOptions extends FetchOptions<OperationExecution> implements Serializable
{
    private static final long serialVersionUID = 1L;

    @JsonProperty
    private PersonFetchOptions owner;

    @JsonProperty
    private OperationExecutionNotificationFetchOptions notification;

    @JsonProperty
    private OperationExecutionSummaryFetchOptions summary;

    @JsonProperty
    private OperationExecutionDetailsFetchOptions details;

    @JsonProperty
    private OperationExecutionSortOptions sort;

    // Method automatically generated with DtoGenerator
    public PersonFetchOptions withOwner()
    {
        if (owner == null)
        {
            owner = new PersonFetchOptions();
        }
        return owner;
    }

    // Method automatically generated with DtoGenerator
    public PersonFetchOptions withOwnerUsing(PersonFetchOptions fetchOptions)
    {
        return owner = fetchOptions;
    }

    // Method automatically generated with DtoGenerator
    public boolean hasOwner()
    {
        return owner != null;
    }

    // Method automatically generated with DtoGenerator
    public OperationExecutionNotificationFetchOptions withNotification()
    {
        if (notification == null)
        {
            notification = new OperationExecutionNotificationFetchOptions();
        }
        return notification;
    }

    // Method automatically generated with DtoGenerator
    public OperationExecutionNotificationFetchOptions withNotificationUsing(OperationExecutionNotificationFetchOptions fetchOptions)
    {
        return notification = fetchOptions;
    }

    // Method automatically generated with DtoGenerator
    public boolean hasNotification()
    {
        return notification != null;
    }

    // Method automatically generated with DtoGenerator
    public OperationExecutionSummaryFetchOptions withSummary()
    {
        if (summary == null)
        {
            summary = new OperationExecutionSummaryFetchOptions();
        }
        return summary;
    }

    // Method automatically generated with DtoGenerator
    public OperationExecutionSummaryFetchOptions withSummaryUsing(OperationExecutionSummaryFetchOptions fetchOptions)
    {
        return summary = fetchOptions;
    }

    // Method automatically generated with DtoGenerator
    public boolean hasSummary()
    {
        return summary != null;
    }

    // Method automatically generated with DtoGenerator
    public OperationExecutionDetailsFetchOptions withDetails()
    {
        if (details == null)
        {
            details = new OperationExecutionDetailsFetchOptions();
        }
        return details;
    }

    // Method automatically generated with DtoGenerator
    public OperationExecutionDetailsFetchOptions withDetailsUsing(OperationExecutionDetailsFetchOptions fetchOptions)
    {
        return details = fetchOptions;
    }

    // Method automatically generated with DtoGenerator
    public boolean hasDetails()
    {
        return details != null;
    }

    // Method automatically generated with DtoGenerator
    @Override
    public OperationExecutionSortOptions sortBy()
    {
        if (sort == null)
        {
            sort = new OperationExecutionSortOptions();
        }
        return sort;
    }

    // Method automatically generated with DtoGenerator
    @Override
    public OperationExecutionSortOptions getSortBy()
    {
        return sort;
    }
    @Override
    protected FetchOptionsToStringBuilder getFetchOptionsStringBuilder()
    {
        FetchOptionsToStringBuilder f = new FetchOptionsToStringBuilder("OperationExecution", this);
        f.addFetchOption("Owner", owner);
        f.addFetchOption("Notification", notification);
        f.addFetchOption("Summary", summary);
        f.addFetchOption("Details", details);
        return f;
    }

}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/operation/fetchoptions/OperationExecutionFetchOptions.js
define([ "require", "stjs", "as/dto/common/fetchoptions/FetchOptions", "as/dto/person/fetchoptions/PersonFetchOptions", "as/dto/operation/fetchoptions/OperationExecutionNotificationFetchOptions",
		"as/dto/operation/fetchoptions/OperationExecutionSummaryFetchOptions", "as/dto/operation/fetchoptions/OperationExecutionDetailsFetchOptions",
		"as/dto/operation/fetchoptions/OperationExecutionSortOptions" ], function(require, stjs, FetchOptions) {
	var OperationExecutionFetchOptions = function() {
	};
	stjs.extend(OperationExecutionFetchOptions, FetchOptions, [ FetchOptions ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.operation.fetchoptions.OperationExecutionFetchOptions';
		constructor.serialVersionUID = 1;
		prototype.owner = null;
		prototype.notification = null;
		prototype.summary = null;
		prototype.details = null;
		prototype.sort = null;

		prototype.withOwner = function() {
			if (this.owner == null) {
				var PersonFetchOptions = require("as/dto/person/fetchoptions/PersonFetchOptions");
				this.owner = new PersonFetchOptions();
			}
			return this.owner;
		};
		prototype.withOwnerUsing = function(fetchOptions) {
			return this.owner = fetchOptions;
		};
		prototype.hasOwner = function() {
			return this.owner != null;
		};

		prototype.withNotification = function() {
			if (this.notification == null) {
				var OperationExecutionNotificationFetchOptions = require("as/dto/operation/fetchoptions/OperationExecutionNotificationFetchOptions");
				this.notification = new OperationExecutionNotificationFetchOptions();
			}
			return this.notification;
		};
		prototype.withNotificationUsing = function(fetchOptions) {
			return this.notification = fetchOptions;
		};
		prototype.hasNotification = function() {
			return this.notification != null;
		};

		prototype.withSummary = function() {
			if (this.summary == null) {
				var OperationExecutionSummaryFetchOptions = require("as/dto/operation/fetchoptions/OperationExecutionSummaryFetchOptions");
				this.summary = new OperationExecutionSummaryFetchOptions();
			}
			return this.summary;
		};
		prototype.withSummaryUsing = function(fetchOptions) {
			return this.summary = fetchOptions;
		};
		prototype.hasSummary = function() {
			return this.summary != null;
		};

		prototype.withDetails = function() {
			if (this.details == null) {
				var OperationExecutionDetailsFetchOptions = require("as/dto/operation/fetchoptions/OperationExecutionDetailsFetchOptions");
				this.details = new OperationExecutionDetailsFetchOptions();
			}
			return this.details;
		};
		prototype.withDetailsUsing = function(fetchOptions) {
			return this.details = fetchOptions;
		};
		prototype.hasDetails = function() {
			return this.details != null;
		};

		prototype.sortBy = function() {
			if (this.sort == null) {
				var OperationExecutionSortOptions = require("as/dto/operation/fetchoptions/OperationExecutionSortOptions");
				this.sort = new OperationExecutionSortOptions();
			}
			return this.sort;
		};
		prototype.getSortBy = function() {
			return this.sort;
		};
	}, {
		owner : "PersonFetchOptions",
		notification : "OperationExecutionNotificationFetchOptions",
		summary : "OperationExecutionSummaryFetchOptions",
		details : "OperationExecutionDetailsFetchOptions",
		sort : "OperationExecutionSortOptions"
	});
	return OperationExecutionFetchOptions;
})